### DISCUSSION TITLE
Rerun a FutureViewModel when a reactive value has changed

### DISCUSSION THREAD

**Discussion by FickleLife** [FilledStacks/stacked#1047] [Q&A]
In my FutureViewModel, I have

```
// reactivity
  @override
  List<ListenableServiceMixin> get listenableServices => [_stateService];

  int get nodeIdSelected => _stateService.nodeId;
```

What I would like to do is listen for the change in `nodeIdSelected` and rerun my `futureToRun()` function in my FutureViewModel to get data relating to the new ID. At the moment, the UI updates fine with the ID itself, so I know the value is updating in the view, but it is not updated in the view model. 

Any clues on how to implement? Thanks

EDIT:

In the old implementation of reactive data, this code would accomplish it but this doesn't work anymore

```
MeasurementDetailModel() {
    // initalise the viewmodel when somethign is added to the stream
    _stateService.streamControllerSearchExecute.stream.listen((value) {
      print('searching...');
      initialise();
    });
```

**Comment by FickleLife** [comment]
Further to this, how would you initialise/refresh data from a single future in a MultipleFutureViewModel ? If you run `initialise`, all Futures re-run.

**Comment by FilledStacks** [comment]
Hey, 

You are correct that you simply have to call initialise again. You can do this by creating a setter for the value you want to react to, and only set the value through the setter function. 

When the value changes you can call initialise. 

In terms of re-running a single future in multi future, that's not a supported functionality at the moment.
